src/Nethermind/Nethermind.Core/Caching/LruKeyCache.cs

```diff
@@ -100,13 +101,12 @@ public void Delete(TKey key)
 
         private void Replace(TKey key)
         {
-            // TODO: some potential null ref issue here?
-            
             LinkedListNode<TKey>? node = _lruList.First;
             _lruList.RemoveFirst();
-            _cacheMap.Remove(node.Value);
+            if (node != null) _cacheMap.Remove(node.Value);
 
-            node.Value = key;
+            // performance optimization: don't create a new node, just re-use the node we just removed
+            node!.Value = key;
```

Comment:
It took me a bit to figure out what was going on here, and I'm pretty sure you are just reusing this object to avoid creating a new instance.  I suspect that is for performance reasons, so I figured I would add a comment to inform the next reader why that re-use is happening.

Reply:
is it recyclable cache? yes, it as a significant performance boost when I found it